File: fyc-api-rest/src/main/java/fr/esgi/fyc_api_rest/rest/SportifRestController.java
package fr.esgi.fyc_api_rest.rest;

import fr.esgi.fyc_api_rest.business.Sportif;
import fr.esgi.fyc_api_rest.dto.request.SportifRequestDTO;
import fr.esgi.fyc_api_rest.dto.response.SportifResponseDTO;
import fr.esgi.fyc_api_rest.mapper.SportifMapper;
import fr.esgi.fyc_api_rest.service.SportifService;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.stream.Collectors;

@RestController
@RequestMapping("/api/Sportifs")
public class SportifRestController {
    private final SportifService sportifService;
    private final SportifMapper sportifMapper;

    public SportifRestController(SportifService sportifService, SportifMapper sportifMapper){
        this.sportifService = sportifService;
        this.sportifMapper = sportifMapper;
    }

    @PostMapping
    @ResponseStatus(code = HttpStatus.CREATED)
    public SportifResponseDTO create(@RequestBody SportifRequestDTO sportifDTO){
        Sportif sportif = sportifService.create(sportifMapper.toEntity(sportifDTO));
        return sportifMapper.toResponseDTO(sportif);
    }

    @GetMapping
    @ResponseStatus(code=HttpStatus.OK)
    public List<SportifResponseDTO> findAll(){
        List<Sportif> sportifs = sportifService.findAll();
        return sportifs.stream().map(sportifMapper::toResponseDTO).toList();
    }

    @GetMapping("/{id}")
    @ResponseStatus(code=HttpStatus.OK)
    public SportifResponseDTO findById(@PathVariable Long id){
        Sportif sportif = sportifService.findById(id);
        return sportifMapper.toResponseDTO(sportif);
    }

    @PutMapping("/{id}")
    @ResponseStatus(code = HttpStatus.OK)
    public SportifResponseDTO update(@PathVariable Long id, @RequestBody SportifRequestDTO sportifCreationDTO){
        Sportif sportif = sportifService.update(id, sportifMapper.toEntity(sportifCreationDTO));
        return sportifMapper.toResponseDTO(sportif);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(code=HttpStatus.NO_CONTENT)
    public void delete(@PathVariable Long id){
        sportifService.delete(id);
    }
}

File: fyc-rest-api-csharp-test/CometeAPI/Rest/SportifRestController.cs
using AutoMapper;
using FycRestApi.Business;
using FycRestApi.DTO.Request;
using FycRestApi.DTO.Response;
using FycRestApi.Service.Interface;
using Microsoft.AspNetCore.Mvc;

namespace FycRestApi.Rest;

[ApiController]
[Route("api/Sportifs")]

public class SportifRestController : ControllerBase
{
    private readonly ISportifService _sportifService;
    private readonly IMapper _mapper;
    public SportifRestController(ISportifService sportifService, IMapper mapper)
    {
        this._sportifService = sportifService;
        this._mapper = mapper;
    }

    [HttpGet("")]
    public async Task<ActionResult<List<SportifResponseDTO>>> Index()
    {
        List<Sportif> sportifs = await this._sportifService.findAll();
        return Ok(_mapper.Map<List<SportifResponseDTO>>(sportifs));
    }

    [HttpPost]
    public async Task<ActionResult<SportifResponseDTO>> Create([FromBody] SportifRequestDTO requestDTO)
    {
        try
        {
            Sportif sportif = await _sportifService.save(_mapper.Map<Sportif>(requestDTO));
            string uri = Url.Action(action: nameof(FindById), new {id = sportif.id});
            return Created(uri, sportif);
        }
        catch (Exception ex) {
            return BadRequest(ex.Message);
        }
    }

    [HttpGet("{id}")]
    public async Task<ActionResult<SportifResponseDTO>> FindById(int id)
    {
        try
        {
            Sportif sportif = await _sportifService.findById(id);
            return Ok(_mapper.Map<SportifResponseDTO>(sportif));
        }
        catch (KeyNotFoundException ex)
        {
            return NotFound(ex.Message);
        }
        catch (Exception ex) {
            return BadRequest(ex.Message);
        }
    }

    [HttpPut("{id}")]
    public async Task<ActionResult<SportifResponseDTO>> update(int id, [FromBody] SportifRequestDTO requestDTO)
    {
        try
        {
            Sportif sportif = await _sportifService.update(id, _mapper.Map<Sportif>(requestDTO));
            return Ok(_mapper.Map<SportifResponseDTO>(sportif));
        }
        catch (KeyNotFoundException ex)
        {
            return NotFound(ex.Message);
        }
        catch (Exception ex)
        {
            return BadRequest(ex.Message);
        }
    }

    [HttpDelete("{id}")]
    public async Task<IActionResult> delete(int id)
    {
        try
        {
            await _sportifService.delete(id);
            return NoContent();
        }
        catch (KeyNotFoundException ex)
        {
            return NotFound(ex.Message);
        }
        catch (Exception ex)
        {
            return BadRequest(ex.Message);
        }
    }
}